File: Prog/Four_prime/Four_Prime.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

bool is_prime(int n) {
    if (n <= 1) {
        return false;
    }
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int* find_four_primes(int n) {
    int* primes = (int*)malloc(4 * sizeof(int));
    for (int i = 2; i < n; i++) {
        if (is_prime(i)) {
            for (int j = i; j < n; j++) {
                if (is_prime(j)) {
                    for (int k = j; k < n; k++) {
                        if (is_prime(k)) {
                            for (int l = k; l < n; l++) {
                                if (is_prime(l) && i + j + k + l == n) {
                                    primes[0] = i;
                                    primes[1] = j;
                                    primes[2] = k;
                                    primes[3] = l;
                                    return primes;
                                }
                            }
                        }
                    }
                }
            }
        }
    }
    free(primes);
    return NULL;
}

int main() {
    int n;
    while (scanf("%d", &n) != EOF) {
        int* result = find_four_primes(n);
        if (result == NULL) {
            printf("Impossible.\n");
        } else {
            printf("%d %d %d %d\n", result[0], result[1], result[2], result[3]);
            free(result);
        }
    }
    return 0;
}

File: Prog/Four_prime/Four_Prime.py
def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

def find_four_primes(n):
    primes = []
    for i in range(2, n):
        if is_prime(i):
            for j in range(i, n):
                if is_prime(j):
                    for k in range(j, n):
                        if is_prime(k):
                            for l in range(k, n):
                                if is_prime(l) and i + j + k + l == n:
                                    primes.append(i)
                                    primes.append(j)
                                    primes.append(k)
                                    primes.append(l)
                                    return primes
    return None

# Read input until end of file
while True:
    try:
        n = int(input())
        result = find_four_primes(n)
        if result is None:
            print("Impossible.")
        else:
            print(" ".join(str(num) for num in result))
    except EOFError:
        break
